src/adapters/openx.js

```diff
@@ -1,130 +1,228 @@
-        scriptUrl = bid.params.jstag_url;
+    var bids = $$PREBID_GLOBAL$$._bidsRequested.find(bidSet => bidSet.bidderCode === 'openx').bids;
+    for (let i = 0; i < bids.length; i++) {
+      let bid = bids[i];
+      let auid = null;
+      let adUnit = null;
+      // find the adunit in the response
+      for (let j = 0; j < adUnits.length; j++) {
+        adUnit = adUnits[j];
+        if (String(bid.params.unit) === String(adUnit.adunitid) && adUnitHasValidSizeFromBid(adUnit,bid) && !adUnit.used) {
+          auid = adUnit.adunitid;
+          break;
+        }
       }
 
-      if (bid.params.pgid) {
-        opts.pgid = bid.params.pgid;
+      let beaconParams = {
+        bd: +(new Date()) - startTime,
+        br: '0', // maybe 0, t, or p
+        bt: $$PREBID_GLOBAL$$.cbTimeout || $$PREBID_GLOBAL$$.bidderTimeout , // For the timeout per bid request
+        bs: window.location.hostname
+      };
+
+      // no fill :(
+      if (!auid) {
+        addBidResponse(null, bid);
+        continue;
       }
+      adUnit.used = true;
+
+      if (adUnit.pub_rev) {
+        beaconParams.br = beaconParams.bt < beaconParams.bd ? 't' : 'p';
+        beaconParams.bp = adUnit.pub_rev;
+        beaconParams.ts = adUnit.ts;
+        addBidResponse(adUnit, bid);
+      } 
+      buildBoPixel(adUnit.creative[0], beaconParams);
+    }
+  };
+
+  function getViewportDimensions(isIfr) {
+    var width,
+      height,
+      tWin = window,
+      tDoc = document,
+      docEl = tDoc.documentElement,
+      body;
+
+    if (isIfr) {
+      tWin = window.top;
+      tDoc = window.top.document;
+      docEl = tDoc.documentElement;
+      body = tDoc.body;
+
+      width = tWin.innerWidth || docEl.clientWidth || body.clientWidth;
+      height = tWin.innerHeight || docEl.clientHeight || body.clientHeight;
+    } else {
+      docEl = tDoc.documentElement;
+      width = tWin.innerWidth || docEl.clientWidth;
+      height = tWin.innerHeight || docEl.clientHeight;
     }
 
-    _requestBids();
+    return `${width}x${height}`;
   }
 
-  function _requestBids() {
+  function makePDCall(pixelsUrl) {
+    let pdFrame = utils.createInvisibleIframe();
+    let name = 'openx-pd';
+    pdFrame.setAttribute("id", name);
+    pdFrame.setAttribute("name", name);
+    let rootNode = document.body;
 
-    if (scriptUrl) {
-      adloader.loadScript(scriptUrl, function () {
-        var i;
-        var POX = OX();
+    if (!rootNode) {
+      return;
+    }
 
-        if (opts.pageURL) {
-          POX.setPageURL(opts.pageURL);
-        }
+    pdFrame.src = pixelsUrl;
 
-        if (opts.refererURL) {
-          POX.setRefererURL(opts.refererURL);
+    if (pdNode) {
+      pdNode.parentNode.replaceChild(pdFrame, pdNode);
+      pdNode = pdFrame;
+    } else {
+      pdNode = rootNode.appendChild(pdFrame);
+    }
+  }
+
+  function addBidResponse(adUnit, bid) {
+    var bidResponse = bidfactory.createBid(adUnit ? CONSTANTS.STATUS.GOOD : CONSTANTS.STATUS.NO_BID);
```

Comment:
Please pass the bidRequest bid object through the `bidfactory.createBid(CONSTANTS.STATUS.GOOD, bid)` so that we can match the `bidId` param. Example https://github.com/prebid/Prebid.js/blob/master/src/adapters/appnexusAst.js#L116